tests/unit/s2n_psk_test.c

```diff
@@ -724,5 +730,190 @@ int main(int argc, char **argv)
         }
     }
 
+    /* Test: s2n_psk_set_hmac */
+    {
+        struct s2n_connection *conn;
+        EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_CLIENT));
+
+        struct s2n_psk *psk = NULL;
+        EXPECT_OK(s2n_array_pushback(&conn->psk_params.psk_list, ( void ** )&psk));
+        uint8_t test_identity[] = "test identity";
+        EXPECT_SUCCESS(s2n_psk_init(psk, S2N_PSK_TYPE_EXTERNAL));
+        EXPECT_SUCCESS(s2n_psk_new_identity(psk, test_identity, sizeof(test_identity)));
+
+        s2n_psk_hmac psk_hmac_alg = -1;
+
+        EXPECT_ERROR_WITH_ERRNO(s2n_psk_set_hmac(psk, psk_hmac_alg), S2N_ERR_HMAC_INVALID_ALGORITHM);
+
+        psk_hmac_alg = S2N_PSK_HMAC_SHA224;
+        EXPECT_OK(s2n_psk_set_hmac(psk, psk_hmac_alg));
+        EXPECT_EQUAL(psk->hmac_alg, S2N_HMAC_SHA224);
+
+        EXPECT_SUCCESS(s2n_connection_free(conn));
+    }
+
+    /* Test setting the callback to select PSK identity */
```

Comment:
Actually, this in accordance with current practice. Callbacks/functions defined in s2n_config.c are not present in s2n_config_test.c but in their relevant feature test files. If we move this to s2n_config_test.c, it is an anti-pattern and we would have to fix the other unit tests as well.

Reply:
I wouldn't call putting the test somewhere discoverable an "anti-pattern." How would someone looking for the tests of this method know to check s2n_psk_test instead of s2n_config_test, when the function is in s2n_config?

Are you sure the function tests being in different files is deliberate and not a result of those functions being moved out of s2n_config.c after the tests were written?

Reply:
Yes, I verified, s2n_config_test does not contain the unit tests, See: https://github.com/awslabs/s2n/blob/main/tests/unit/s2n_config_test.c